core/cat/db/crud.py

```diff
@@ -69,4 +69,7 @@ def upsert_setting_by_name(payload: models.Setting) -> models.Setting:
         query = Query()
         get_db().update(payload, query.name == payload.name)
 
-    return get_setting_by_name(payload.name)
\ No newline at end of file
+    return get_setting_by_name(payload.name)
+
+def validate_presences(required: list, payload: dict):
```

Comment:
Agree with the utility of a `validate_presences` method, but it is not related to DB crud.
I would move it here in the [factory](https://github.com/cheshire-cat-ai/core/tree/a0186f5559b4557c3f9bab015eaa2d2e0e1620cf/core/cat/factory), where schemas are defined, or in [utilis](https://github.com/cheshire-cat-ai/core/blob/a0186f5559b4557c3f9bab015eaa2d2e0e1620cf/core/cat/utils.py) in case we'll use this validation method also for plugin settings. Also [infrastructure](https://github.com/cheshire-cat-ai/core/tree/a0186f5559b4557c3f9bab015eaa2d2e0e1620cf/core/cat/infrastructure) could be a place for this method, if we go for a validation set of methods

What do you think?